src/atom.rs

```diff
@@ -67,8 +67,13 @@ impl StringCacheEntry {
 
 impl StringCache {
     fn new() -> StringCache {
+        type T = Option<Box<StringCacheEntry>>;
+        assert!(std::mem::size_of::<T>() == std::mem::size_of::<usize>());
```

Comment:
This can be checked statically with `std::mem::transmute`.

Reply:
Done